### PR TITLE
aic-emu: Support for Resolution Change

### PR DESCRIPTION
**Pull request by sandeep-maddipatla**
This commit also supports cases where the test content supplied doesn't match the input profile resolution. 
- This is required to support tests where the game resolution changes
- Note that the app still expects the test clip to be larger than or equal to what the profile requires

A minor refactoring is done to support this: Moved surface parameter initialization into the CmdHandler class from the GfxHandler class, due to the former class having better access to the required data (width and stride parameters)

Related Jira: VSMGWL-51998

### CODE REVIEW

File: aic-emu/CmdHandler.cc [hunk 1]
```diff
@@ -643,3 +644,33 @@ int CmdHandler::GetOneFrame(uint8_t* pFrame, SurfaceParams_t surf)
...
+    surf.tilingFormat  = m_props->tiling_mode; //will be overridden by adjust function
+    surf.alignedWidth  = m_props->aligned_width;
+    surf.alignedHeight = m_props->aligned_height;
+
+    //Adjust params as required by gfx driver
+    int sts = m_gfx.AdjustSurfaceParams(surf);
+
+    if (sts)
+        return AICS_ERR_GFX;
+	else
```
Comment (dvrogozh) [inline aic-emu/CmdHandler.cc:674]: remove tab

**Review by dvrogozh — CHANGES_REQUESTED** [review]
minor changes suggested/requested

File: aic-emu/CmdHandler.cc (hunk 1 above)
Comment (sandeep-maddipatla) [reply aic-emu/CmdHandler.cc]: done

File: aic-emu/CmdHandler.cc [hunk 2]
```diff
@@ -560,25 +579,67 @@ int CmdHandler::InitGfxSurfaceHandler()
...
+                  << "is smaller than Gfx surface dimensions("
+                  << surf.width << "x" << surf.height << ")" << std::endl;
+
+        return AICS_ERR_STREAM_PARAMS;
+    }
+
+    unsigned srcStride = m_srcWidth * surf.pixelSize; //Content file
+    unsigned dstStride = surf.width * surf.pixelSize; //CPU Buffer to copy to gfx surface
+
+    auto src = new uint8_t[srcStride];
```
[the remark this answers is not in the document]
Comment (sandeep-maddipatla) [reply aic-emu/CmdHandler.cc:627]: I think a memory blob is more suitable here, as we are using this structure for copy ops. We are deallocating it in the same function, so it should be safe.

[the remark this answers is not in the document]
Comment (dvrogozh) [reply aic-emu/CmdHandler.cc:627]: ok, that's not critical

**Review by dvrogozh — APPROVED** [review]
looks good

**Pull request merged**